### ISSUE TITLE
[Regression] Navigation: Blender style - unintuitive camera behavior when rotation center is far from the object

### ISSUE THREAD

**Issue by FEA-eng** [FreeCAD/FreeCAD#17619]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Everything is described in the forum thread but here's the most important part: https://forum.freecad.org/viewtopic.php?t=91595&start=10#p789527

In short, #17341 fixed some rather small bug in navigation but also brought a serious regression to the Blender style. Fortunately, #17466 mostly fixed it but there still seems to be some important regression reported in the linked forum post and confirmed by Werner: https://forum.freecad.org/viewtopic.php?t=91595&start=20#p789529

Basically, camera rotation is inconvenient when the rotation center is far from the object.

### Full version info

```shell
OS: Windows 10 build 19045
Architecture: x86_64
Version: 1.1.0dev.39100 (Git)
Build type: Release
Branch: main
Hash: 8865450a3e14220925e0e449c0f1f79056b4fb89
Python 3.11.10, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: Polish/Poland (pl_PL)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
```

### Subproject(s) affected?

Core

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
CC @Rexbas (can you take a look at this?)

**Comment by Rexbas** [comment]
I am already working on this, will have a PR today. And it is not just the Blender style, but all styles combined with drag at cursor mode.

**Comment by adrianinsaval** [comment]
please confirm that the behavior is as desired now

**Comment by Rexbas** [comment]
We have a confirmation on the forum: https://forum.freecad.org/viewtopic.php?p=790355#p790355
